### PR TITLE
Unsupported static image format

### PR DESCRIPTION
**Pull request by oleurud**
Fix: https://github.com/CartoDB/Windshaft/issues/331

### CODE REVIEW

File: lib/cartodb/api/middlewares/check-static-image-format.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const VALID_IMAGE_FORMATS = ['png', 'jpg', 'jpeg'];
+
+module.exports = function checkStaticImageFormat () {
+    return function checkStaticImageFormatMiddleware (req, res, next) {
+        if(!VALID_IMAGE_FORMATS.includes(req.params.format.toLowerCase())) {
```
Comment (enekid) [inline lib/cartodb/api/middlewares/check-static-image-format.js]: Should be so polite?

In the [docs](https://carto.com/developers/maps-api/guides/static-maps-API/#zoom--center) we offer just 2 options: `png` and `jpg`

**Comment by dgaubert** [comment]
Please, resolve conflicts before proceeding :)

**Comment by oleurud** [comment]
I will need to add the merge changes to it.

File: lib/cartodb/api/map/preview-layergroup-controller.js [hunk 2]
```diff
@@ -48,29 +48,15 @@ module.exports = class PreviewLayergroupController {
...
-            ),
-            getPreviewImageByCenter(this.previewBackend),
-            cacheControlHeader(),
-            cacheChannelHeader(),
-            surrogateKeyHeader({ surrogateKeysCache: this.surrogateKeysCache }),
-            lastModifiedHeader()
-        );
+        let getPreviewImage;
+
+        if (previewType === 'centered') {
```
Comment (enekid) [inline lib/cartodb/api/map/preview-layergroup-controller.js]: I don't like `if`s:

```javascript
const previewers = {
   'centered': getPreviewImageByCenter,
   'bbox': getPreviewImageByBoundingBox
}

...

const getPreviewImage = previewers[previewType];
```

Comment (oleurud) [reply lib/cartodb/api/map/preview-layergroup-controller.js]: I agree, but it is another task ;) I mean, I like your solution, but I would like to keep the PRs small and focused

**Pull request merged**